packages/alfa-rules/src/common/expectation/contrast.ts

```diff
@@ -223,3 +230,42 @@ export function contrast(foreground: RGB, background: RGB): number {
 
   return round(contrast * 100) / 100;
 }
+
+/**
+ * Finds elements from a collection of candidate that intersect with a given element
+ *
+ * @remarks
+ * If `candidates` is non-empty and just one element doesn't have layout `None` is returned
```

Comment:
```suggestion
 * If either the element or one of the `candidates` doesn't have layout, we can't fully decide intersection and return `None`.
```
**Note:** it almost feels like this should return a `Result<Iterable<Element>, string>` rather than an `Option` 🤔

Reply:
I'll add a note about it on https://github.com/Siteimprove/alfa/issues/1037